src/Processors/SuggestionsProcessor.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace SilverStripe\DiscovererElasticEnterprise\Processors;
+
+use Exception;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Discoverer\Service\Results\Suggestions;
+
+class SuggestionsProcessor
+{
+
+    use Injectable;
+
+    public function getProcessedSuggestions(Suggestions $suggestions, array $response): void
+    {
+        // Check that we have all critical fields in our Elastic response
+        $this->validateResponse($response);
+
+        $documentSuggestions = $response['results']['documents'] ?? [];
```

Comment:
we're not checking for 'documents' is it possible we'll get an array key error here without it?

Reply:
Null coalescing operator (`??`) supports missing keys for arrays and objects. It will default to `[]` if anything is `null` along the way.